File: 序号/1.两数之和.cpp
class Solution {
 public:
  vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> m;
    int n = nums.size();
    for (int i = 0; i < n; i++) {
      auto ret = m.find(target - nums[i]);
      if (ret == m.end()) {
        m.try_emplace(nums[i],i);
      } else {
        return {i, ret->second};
      }
    }
    return {};
  }
};

File: 序号/1. 两数之和.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d= {}
        for i,x in enumerate(nums):
            if target - x in d:
                return [d[target-x],i]
            d[x]=i
        return [0,0]